### ISSUE TITLE
[K21] Playing DASH (.mpd) Stream stops after some seconds and fall in to buffer loading forever 

### ISSUE THREAD

**Issue by werimail** [peak3d/inputstream.adaptive#1461]
### Describe the bug
Here is a clear and concise description of what the problem is:
I have played http://mcdn.daserste.de/daserste/dash2/manifest.mpd with this stream on my test playback works for only 
some seconds then fall in to buffer loading forever.

playing on Kodi 20 playback seem to works, but instead to get new segments from manifest, insert them manually, but sound strange because there are segments

from what i see also if i try sync in advance the manifest update, its not able to find next segment to be loaded/played and so freeze the playback atm i havent found the reason of this problem but i dont see realations to the current fix PR#1459

## Expected Behavior
Here is a clear and concise description of what was expected to happen:
i can play the stream http://mcdn.daserste.de/daserste/dash2/manifest.mpd without fall in to buffer loading forever

### To Reproduce

1. 
2.
3.

### Debuglog

The debuglog can be found here:

### MPD/M3U8s/ISM

An example or copy of a manifest (or manifests for HLS - master and variants) can be found here:

### Screenshots 
Here are some links or screenshots to help explain the problem:

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

### Your Environment
Used Operating system:

 - [ x] Android
 - [ ] iOS
 - [ ] tvOS
 - [ ] Linux
 - [ ] OSX
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: LineageOS 19 based on Android for Raspberry Pi 400

-   Kodi version: 21 Beta2
 - inputstream.adaptive version: 21.4.1

*note: Once the issue is made we require you to update it with new information should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by werimail** [comment]
Hi @CastagnaIT as requested, here is the new issue for the problem described here: https://github.com/xbmc/inputstream.adaptive/issues/1109#issuecomment-1910790656

**Comment by MauriceW67** [comment]
@werimail You could try one of the latest Jenkins builds from here:

https://jenkins.kodi.tv/blue/organizations/jenkins/xbmc%2Finputstream.adaptive/activity/

You can always go back to the official release by manually updating again from within Kodi.

**Comment by werimail** [comment]
Thanks @MauriceW67 unfortunately there is no solution to the problem yet. The current versions of the Inputstream Adaptive Addons from the named source also throw the same error. But I'm sure @CastagnaIT  is already working on a solution ;-) 

Greetings Weri

**Comment by CastagnaIT** [comment]
im working on a fix
you can find some test builds here:
https://jenkins.kodi.tv/blue/organizations/jenkins/xbmc%2Finputstream.adaptive/detail/PR-1492/4/artifacts
if you can please give a feedback

**Comment by gandalth** [comment]
I am testing the issue posted by linqcan.
Can I use any of your testbuilds on RPi4?

**Comment by CastagnaIT** [comment]
> I am testing the issue posted by linqcan. Can I use any of your testbuilds on RPi4?

if you are using android on RPI, pick android build,
if you are using RPI OS or LE/CE, no, you need to build ISA yourself we dont provide builds for these systems directly

**Comment by gandalth** [comment]
Thank you. I will check the build instructions and do some testing.

**Comment by werimail** [comment]
Hi @CastagnaIT  I looked at the test builds in my Android 13 RPI400 with aarch64 architecture environment. The DASH (.mpd) streams are now running again. Freezing or buffering no longer takes place.

However, the problem with the skipped subtitles described here: https://github.com/xbmc/inputstream.adaptive/issues/1109#issuecomment-1905785094 still exists. Should I also create a separate issue for this problem?

Thanks and greetings Weri

**Comment by CastagnaIT** [comment]
i have tested more times that mpd stream but i can see always subtitles displayed
its not clear to me what means exactly for you "skipped subtitles"
however that Issue was for HLS, yes better separate Issue
and a way to reproduce the problem because until now with my playback tests seem ok, or i missed the real point

if subtitles no longer display during playback also make sure Kodi itself has not disabled them during playback
because there is currently a bug in kodi that can disable sebtitle setting without asking to user, this for adaptive streaming like this one

**Comment by werimail** [comment]
> i have tested more times that mpd stream but i can see always subtitles displayed its not clear to me what means exactly for you "skipped subtitles" however that Issue was for HLS, yes better separate Issue and a way to reproduce the problem because until now with my playback tests seem ok, or i missed the real point
> 
> if subtitles no longer display during playback also make sure Kodi itself has not disabled them during playback because there is currently a bug in kodi that can disable sebtitle setting without asking to user, this for adaptive streaming like this one

Ok @CastagnaIT, then I'll try to formulate the problem with the subtitles a little differently: 
The subtitles are also displayed in the Kodi OSD after activation. However, only one line of text. The remaining lines are then “swallowed”. The sentences shown no longer make sense. In the end, entire parts are skipped as soon as new ones are added. If I access the same stream under Kodi 20.x and also activate the subtitles there, the display is correct even across several lines. Here you can also compare what would have to be output as text for subtitles in contrast to Kodi 21

Greetings Weri

**Comment by CastagnaIT** [comment]
now i think to have understood better, i will try check

**Comment by werimail** [comment]
> now i think to have understood better, i will try check

Many thanks @CastagnaIT

**Comment by CastagnaIT** [comment]
@werimail was an easy fix, done on #1496

**Comment by werimail** [comment]
> @werimail was an easy fix, done on #1496

Oh @CastagnaIT that doesn't sound bad. Is there already a test build to try the fixes #1492 and #1496 together out?

Thanks and greetings Weri

**Comment by CastagnaIT** [comment]
to find a test build in a PR:
open "all checks passed" section and click on "details" link of the "continuous integration" action
![308064493-a407dbd7-9778-4a05-a323-f23e5de1bd7a](https://github.com/xbmc/inputstream.adaptive/assets/3257156/3f3dfdc1-ae55-49c3-9d79-993b16c88d90)
after opened the new page goto "artifact" page to get the test builds, link on top menu

atm not possible both PR together, its needed to wait the merge of PR 1496,
after that i can rebase the PR 1492 that so will include the other PR synched from master branch

**Comment by werimail** [comment]
> to find a test build in a PR: open "all checks passed" section and click on "details" link of the "continuous integration" action ![308064493-a407dbd7-9778-4a05-a323-f23e5de1bd7a](https://private-user-images.githubusercontent.com/3257156/308066785-3f3dfdc1-ae55-49c3-9d79-993b16c88d90.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDkwMTc2NjAsIm5iZiI6MTcwOTAxNzM2MCwicGF0aCI6Ii8zMjU3MTU2LzMwODA2Njc4NS0zZjNkZmRjMS1hZTU1LTQ5YzMtOWQ3OS05OTNiMTZjODhkOTAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDIyNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAyMjdUMDcwMjQwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MWVkNzQ2MmEzZTk3MzBjZjA2NWFjZGJmNTRjMjVkZmFiMzNhMzgyZDY5MDJmZTM1ODBmOGEwMTEzN2Y1Yjc4NiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.ZCF0h35Raay_yfcMJlKaQqh5fSrqIlxgS5YMwLH2foE) after opened the new page goto "artifact" page to get the test builds, link on top menu
> 
> atm not possible both PR together, its needed to wait the merge of PR 1496, after that i can rebase the PR 1492 that so will include the other PR synched from master branch

Thank you @CastagnaIT i will try.

**Comment by werimail** [comment]
I tested the PR #1496 with various DASH (.mpd.) streams. The subtitles are now output correctly again. Even with other Kodi addons (e.g. Zattoo HIQ or PrimeVideo).

The problem:
> if subtitles no longer display during playback also make sure Kodi itself has not disabled them during playback because there is currently a bug in kodi that can disable sebtitle setting without asking to user, this for adaptive streaming like this one

I particularly noticed this with the PrimeVideo addon. Only after the subtitles have been manually activated again will they be output again until Kodi automatically deactivates them again??? Is there already an issue/fix for this?

Greetings Weri

**Comment by CastagnaIT** [comment]
> I particularly noticed this with the PrimeVideo addon. Only after the subtitles have been manually activated again will they be output again until Kodi automatically deactivates them again??? Is there already an issue/fix for this?

atm no, couple of years ago i tried expose the problem to xbmc repo since its a Kodi core bug, but it was very problematic to have someone from the team to help me with this, anyway from my memories the problem should be on `CVideoPlayer::OpenDefaultStreams` in the `PredicateAudioFilter` code that "reset" the audio settings instead of take care of try preserve settings:

https://github.com/xbmc/xbmc/blob/213cb317db5687479e5cf1b4b5058082a6ddadb1/xbmc/cores/VideoPlayer/VideoPlayer.cpp#L904-L905

and above linked code is fired each time DMX_SPECIALID_STREAMCHANGE event happens, from here:
https://github.com/xbmc/xbmc/blob/master/xbmc/cores/VideoPlayer/VideoPlayer.cpp#L1054-L1055
and so each time when ISA change stream chapter, or when change video quality (adaptive streaming)

maybe if someone else highlights the problem by open an Issue on xbmc repo,
probably over time maybe someone takes interest in investigating

**Comment by werimail** [comment]
> > I particularly noticed this with the PrimeVideo addon. Only after the subtitles have been manually activated again will they be output again until Kodi automatically deactivates them again??? Is there already an issue/fix for this?
> 
> atm no, couple of years ago i tried expose the problem to xbmc repo since its a Kodi core bug, but it was very problematic to have someone from the team to help me with this, anyway from my memories the problem should be on `CVideoPlayer::OpenDefaultStreams` in the `PredicateAudioFilter` code that "reset" the audio settings instead of take care of try preserve settings:
> 
> https://github.com/xbmc/xbmc/blob/213cb317db5687479e5cf1b4b5058082a6ddadb1/xbmc/cores/VideoPlayer/VideoPlayer.cpp#L904-L905
> 
> and above linked code is fired each time DMX_SPECIALID_STREAMCHANGE event happens, from here: https://github.com/xbmc/xbmc/blob/master/xbmc/cores/VideoPlayer/VideoPlayer.cpp#L1054-L1055 and so each time when ISA change stream chapter, or when change video quality (adaptive streaming)
> 
> maybe if someone else highlights the problem by open an Issue on xbmc repo, probably over time maybe someone takes interest in investigating

Okay, I have created a new issue with a reference to your comment here. It would be nice if you could add the relevant insights there. Unfortunately, I don't know much about the topic in detail.

Greetings Weri

**Comment by dollhaus** [comment]
Did this change ever make it to the v20 version of inputstream-adaptive? If so, is there a specific jenkins build I should be looking at? The one linked above has no artifacts attached to it

**Comment by CastagnaIT** [comment]
nope only kodi 21,
developments on Kodi 20 are about to be stopped

**Comment by dollhaus** [comment]
I'm presently unable to upgrade to v21 for compatibility reasons, is it theoretically possible to build the v21 addon for v20?

**Comment by CastagnaIT** [comment]
no, are different interfaces
